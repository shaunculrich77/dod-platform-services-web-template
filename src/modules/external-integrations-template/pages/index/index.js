import TechLogos from '@/modules/external-integrations-template/components/tech-logos/tech-logos.vue';

export default {
    name: 'Home',
    components: {
        TechLogos,
    },
};
