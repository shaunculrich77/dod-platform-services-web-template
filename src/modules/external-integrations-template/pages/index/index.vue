<template>
    <div class="v-cloak d-flex flex-column align-items-center">
        <img
            class="img-fluid dod-logo"
            alt="Doctor on Deman"
            src="@/assets/DoD_logo.png"
        />
        <h2 class="p-4">External Integrations Web Template</h2>
        <div>
            <tech-logos></tech-logos>
        </div>
    </div>
</template>

<script lang="js" src="./index.js"></script>
<style lang="scss" src="./index.scss"></style>
