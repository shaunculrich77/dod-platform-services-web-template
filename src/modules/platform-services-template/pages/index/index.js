import TechLogos from '@/modules/platform-services-template/components/tech-logos/tech-logos.vue';

export default {
    name: 'Home',
    components: {
        TechLogos,
    },
    data() {
        return {
            title: 'External Integrations Web Template',
        };
    },
    methods: {
        sum(a, b) {
            return a + b;
        },
    },
};

/*
Naeem Smith: no, you don't need a main.js in the module directories
Naeem Smith: however, there should be an entry file
Naeem Smith: the file that communicates with the outside world
Naeem Smith: you can call it index.js so it denotes that it is the entry into that module
Naeem Smith: any props that is passed into that module should go through that index
Naeem Smith: and any event changes, should go out of that index

but one thing about the entry file. It will be tempting to want to have another 
file receive props. Like directly expose one of your module components and pass props into it. 
I would say also think of the module as an interface with public and private components. 
The only component that is public is the index.js, and if you want to pass props to one of the 
module's components, then you have to go through the public index.js file, and then pass it 
freely to its private components.


*/
