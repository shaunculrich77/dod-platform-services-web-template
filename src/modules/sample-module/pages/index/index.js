export default {
    name: 'sample-index',
};
