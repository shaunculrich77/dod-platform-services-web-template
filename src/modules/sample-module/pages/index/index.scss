.dod-logo {
    width: 15%;
    height: auto;
}
