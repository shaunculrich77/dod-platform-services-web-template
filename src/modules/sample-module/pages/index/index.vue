<template>
    <div>Index for Sample Module using module level router</div>
</template>

<script lang="js" src="./index.js"></script>

<style lang="scss" src="./index.scss"></style>
