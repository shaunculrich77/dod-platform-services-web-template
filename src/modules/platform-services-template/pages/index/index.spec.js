// const { TestScheduler } = require("jest");
import index from '@/modules/platform-services-template/pages/index/index.vue';
import { mount } from '@vue/test-utils';

test('mount a vue component', () => {
    const wrapper = mount(index);
    expect(wrapper.html()).toMatchSnapshot();
});
