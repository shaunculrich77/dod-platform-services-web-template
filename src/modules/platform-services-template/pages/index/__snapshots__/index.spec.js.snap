// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount a vue component 1`] = `
<div class="v-cloak d-flex flex-column align-items-center"><img alt="Doctor on Deman" src="@/assets/DoD_logo.png" class="img-fluid dod-logo">
  <h2 class="p-4">External Integrations Web Template</h2>
  <div>
    <div class="d-flex flex-row images-divider">
      <div class="px-4"><img src="@/assets/logo.png" alt="Vue.js logo" title="Vue.js"></div>
      <div class="px-4"><img src="@/assets/kotlin.png" alt="KMM" title="Kotlin Multiplatform Mobile"></div>
      <div class="px-4"><img src="@/assets/JavaScript-logo.png" alt="JavaScript" title="JavaScript"></div>
      <div class="px-4"><img src="@/assets/GitHub-Mark.png" alt="GitHub" title="GitHub"></div>
      <div class="px-4"><img src="@/assets/vertical-logo-monochromatic.png" alt="Docker" title="Docker"></div>
      <div class="px-4"><img src="@/assets/bootstrap.png" alt="Bootstrap" title="Bootstrap"></div>
      <div class="px-4"><img src="@/assets/jest.jpeg" alt="Jest" title="Jest"></div>
    </div>
  </div>
</div>
`;
